### PR TITLE
Add --fastest and --shortest options to `next`

### PR DESCRIPTION
**Pull request by daxida**

### CODE REVIEW

**Comment by daxida** [comment]
Personally I like the idea of adding the shortcuts -r -s -f to the respective reverse, shortest, fastest modes so one could fast type its desired combination. They don't conflict with anything (at the moment), and the user should be already acquainted with the terms if he/she actually plays clash of code.

File: src/main.rs [hunk 1]
```diff
@@ -259,16 +265,20 @@ impl App {
...
                     for _i in 0..max_attemps {
                         let handle = self.random_handle()?;
                         let clash = self.read_clash(&handle)?;
-                        if clash.is_reverse() {
-                            return Ok(handle);
-                        }
+                        if reverse  && !clash.is_reverse()  { continue; }
+                        if fastest  && !clash.is_fastest()  { continue; }
+                        if shortest && !clash.is_shortest() { continue; }
+                        return Ok(handle);
```
Comment (Andriamanitra) [inline src/main.rs]: I think the use of continue here is a bit weird, why not make it into a single if?
```rust
if (!reverse || clash.is_reverse()) && (!fastest || clash.is_fastest()) && (!shortest || clash.is_shortest()) {
    return Ok(handle);
}
```

File: src/main.rs [hunk 2]
```diff
@@ -259,16 +265,20 @@ impl App {
...
                     }
-                    Err(anyhow!(format!("No reverse clash found after {} attempts.", max_attemps)))
+                    Err(anyhow!(format!("No required clash found after {} attempts.", max_attemps)))
```
Comment (Andriamanitra) [inline src/main.rs]: The wording "required clash" is a bit confusing. Maybe something like "Failed to find a clash with the required modes" instead?

File: src/main.rs (hunk 1 above)
Comment (daxida) [reply src/main.rs]: I find that long if harder to parse personally (even if it is the same thing).

**Pull request merged**